lib/semanticchess.js

```diff
@@ -191,6 +191,39 @@ class SemanticChess {
     }
   }
 
+  /**
+   * This method evaluates a move to check if it is a pawn promotion
+   * For that it must:
+   * - Be valid
+   * - Be a pawn
+   * - Move onto rank 8 or 1 from the previous ones
+   * @param {string} source Position of the move's start 
+   * @param {string} target Position of the move's destination
+   */
+  isPromotion(source, target){
+    var moveToTest = {
+      from: source,
+      to: target,
+      promotion :'q'
+    };
+
+    var move = this.chess.move(moveToTest);
+    if (move === null)
+      return false;
+    else
+      this.chess.undo();
+    
+    var rankSource = source.substring(2,1);
+    var rankTarget = target.substring(2,1);
+    var piece = this.chess.get(source).type;
+
+    if (piece === 'p' &&
```

Comment:
This `if` together with the `return false` could all be turned into one `return`.